File: UnitTesting/SpecFlow-Examples-master/BowlingKata/BowlingKata-XUnit/Bowling.Specflow/Drivers/BowlingDriver.cs
using TechTalk.SpecFlow;
using Xunit;

namespace Bowling.Specflow.Drivers
{
    public class BowlingDriver
    {
        private Game _game;

        public void NewGame()
        {
            _game = new Game();
        }

        public void Roll(int pins, int rollCont)
        {
            for (int i = 0; i < rollCont; i++)
            {
                _game.Roll(pins);
            }
        }

        public void Roll(int pins1, int pins2, int rollCount)
        {
            for (int i = 0; i < rollCount; i++)
            {
                _game.Roll(pins1);
                _game.Roll(pins2);
            }
        }

        public void RollSeries(string series)
        {
            foreach (string roll in series.Trim().Split(','))
            {
                _game.Roll(int.Parse(roll));
            }
        }

        public void RollSeries(Table rolls)
        {
            foreach (var row in rolls.Rows)
            {
                _game.Roll(int.Parse(row["Pins"]));
            }
        }

        public void CheckScore(int expected)
        {
            Assert.Equal(expected, _game.Score);
        }
    }
}

File: UnitTesting/Old/SpecFlow-Examples-master/BowlingKata/BowlingKata-VB-MsTest/Bowling.Specflow/Drivers/BowlingDriver.vb
Imports Bowling.Bowling
Imports TechTalk.SpecFlow

Namespace Drivers
    Public Class BowlingDriver
        Private _game As Game

        Public Sub NewGame()
            _game = New Game()
        End Sub

        Public Sub Roll(pins As Integer, rollCount As Integer)
            For i = 1 To rollCount
                _game.Roll(pins)
            Next
        End Sub

        Public Sub Roll(pins1 As Integer, pins2 As Integer, rollCount As Integer)
            For i = 1 To rollCount
                _game.Roll(pins1)
                _game.Roll(pins2)
            Next
        End Sub

        Public Sub RollSeries(series As String)
            For Each roll As String In series.Trim().Split(",")
                _game.Roll(Integer.Parse(roll))
            Next
        End Sub

        Public Sub RollSeries(rolls As Table)
            For Each row As TableRow In rolls.Rows
                _game.Roll(Integer.Parse(row.Item("Pins")))
            Next
        End Sub

        Public Sub CheckScore(expected As Integer)
            Assert.AreEqual(expected, _game.Score)
        End Sub
    End Class
End Namespace